### ISSUE TITLE
11ty suggestion

### ISSUE THREAD

**Issue by Archeelux** [Miffyli/sym#44]
Hi,

I love the website and the data it provides. As soon as I seen github I wanted to see if I can contribute in any way.

I think this website would heavily benefit from static site generator (SSG), my suggestion would be to use [11ty](https://www.11ty.io/). 

What do you guys think? It would allow you to re use parts of html using templates and among other things will generate html output depending on what you want, while keeping the project DRY.

**Comment by Miffyli** [comment]
Hey! I am no professional web-developer so excuse my possibly dumb comments, but I understood this would generate some kind of template representation into static HTML pages which are then served to the user?

We originally picked "pure JS" for few reasons: 
1) Very simple for server (No PHP. Only static content)
2) No reliance on frameworks to avoid issues from there
3) Bit smoother usage compared to loading different pages (think: old Symthic)

Sounds like this would present a framework and then bit of work from building the static content whenever things change? Admittedly, we used template type of thing back with original Symthic, which served us quite well for being an in-house engine behind the scenes. 

We are still ironing out bunch of bigger issues, but this could be considered depending on how big job it would be to change to this. @robenter Could you give comments on this, as you are the main web guy?

**Comment by robenter** [comment]
I think using something like SSG would be more beneficial to sites like blogs or news site that are generating tons of content (multiple articles/posts a day) or sites that present a database full of data (i.e. Wikipedia).  The Sym page is pretty static as far as content and doesn't need that level of automation.  Not to mention, the content that is presented is usually presented in a way that is unique to that content (ie. weapon chart, weapon comparison, equipment, general info, etc..)  IMHO it would be way overkill to use a framework like this.

**Comment by Archeelux** [comment]
I understand your issue with maybe it being an overkill but SSG in general is the future. Especially the JAMStack approach that has become pretty prevalent lately. Just looking through the repo it self I can see that [https://github.com/Miffyli/sym/blob/master/webpage/pages/bfv/bfv_dataWeapon.html](bfv_dataWeapon.html) has a lot of repeating code in it.

For example using 11ty you can easily create a component and store all the data in the JSON format to repeat seactions of html, which all gets generated in a static file at the end of the day. 

e.g. (from my own project to display multiple images)
```
<div class="columns is-multiline is-mobile">
    {% for img in collections.portfolio %}
        <div class="column is-3-desktop is-4-tablet is-6-mobile image-card">
            <a href="{{ img.url }}" class="image-link">
                <figure class="image is-square is-fullwidth">
                    <img src="{{img.data.image}}?nf_resize=fit&w=480&h=480" alt="image of '{{img.data.title}}'">
                </figure>
                <div class="card">
                    <h2 class="title has-text-white">{{img.data.title}}</h2>
                    <h3 class="subtitle has-text-light">
                        Find out more
                    </h3>
                </div>
            </a>
        </div>
    {% endfor %}
</div>
```

I think it can help with keeping the website clean and fast for both end user and the developer. It is up to you of course and I am just giving my 2 cents.

**Comment by Miffyli** [comment]
Code duplication is (almost) by design: With Symthic we learned that even different titles of same series of games can have small differences that make sharing code like this difficult, up to a point we had to start duplicating code to avoid overly complex functions. The page you refer to is one of the "document" type which is not designed to be compact, rather a static, hand-written document. Ideally that would be written in Markdown or similar markup language, but for now we are going with raw HTML. 

We will move considering frameworks and bigger changes after most of the site works as expected :)
